### ISSUE TITLE
No symlinks, is this the expected behavior ?

### ISSUE THREAD

**Issue by LaMarMot** [graysky2/anything-sync-daemon#87]
</div>
            <div class='post' itemprop='text'>
              <p>Hey</p>
<p>So, I’m using <a href="https://wiki.archlinux.org/index.php/Anything-sync-daemon" rel="noopener nofollow ugc"><code>anything-sync-daemon</code></a> (<a href="https://archlinux.org/packages/?name=anything-sync-daemon" rel="noopener nofollow ugc">asd</a>) on Arch Linux (EndeavourOS install to be exact) and I’m just wondering if it’s working as it should.</p>
<p>I’m using it to manage the <code>chromium</code> cache directory.<br>
It’s suppose to “move” this directory in <a href="https://wiki.archlinux.org/title/Tmpfs" rel="noopener nofollow ugc"><code>tmpfs</code></a> (i.e. in RAM) and synchronize it to the hard drive from time to time…</p>
<p>So when I “ask” it if everything is going according to plan, it seems to tell me “yes”:</p>
<pre><code class="lang-auto">$ asd p
Waiting for lock...
flock: getting lock took 0.000002 seconds
flock: executing /usr/bin/asd
Anything-sync-daemon v6.0.0 on EndeavourOS

 Systemd service is currently active.
 Systemd resync service is currently active.
 Overlayfs v23 is currently active.

Asd will manage the following per /run/asd.conf settings:

 owner/group id:     user/1000
 target to manage:   /home/user/.cache/chromium
 sync target:        /home/user/.cache/.chromium-backup_asd
 tmpfs target:       /tmp/asd-user/home/user/.cache/chromium
 dir size:           421M
 overlayfs size:     280K
 recovery dirs:      5 &lt;- delete with the c option
  dir path/size:     /home/user/.cache/.chromium-backup_asd-crashrecovery-20240210_204103.tar.zstd (267M)
  dir path/size:     /home/user/.cache/.chromium-backup_asd-crashrecovery-20240128_121006.tar.zstd (4,0K)
  dir path/size:     /home/user/.cache/.chromium-backup_asd-crashrecovery-20240128_114248.tar.zstd (8,6M)
  dir path/size:     /home/user/.cache/.chromium-backup_asd-crashrecovery-20240116_191014.tar.zstd (298M)
  dir path/size:     /home/user/.cache/.chromium-backup_asd-crashrecovery-20240111_141324.tar.zstd (312M)
</code></pre>
<p>But if I look at the “managed” folder ( <code>/home/user/.cache/chromium</code>) it is still a “regular” directory, shouldn’t it rather be a <strong>sym</strong>bolic <strong>link</strong> to <code>/tmp/asd-user/home/user/.cache/chromium</code>?</p>
<p>I’m using <a href="https://archlinux.org/packages/?name=profile-sync-daemon" rel="noopener nofollow ugc">profile-sync-daemon</a> (<a href="https://wiki.archlinux.org/title/Profile-sync-daemon" rel="noopener nofollow ugc">psd</a>) to achieve the same behavior with my browsers profiles, and, in that case, the managed directories are (became) indeed <strong>symlinks</strong> to the directories in tmpfs…</p>
<p>Have I done something wrong or is <code>asd</code> working as it should (i.e. not writing cache on disk anymore, except to back it up)?</p>
<p>I would be very grateful for any advice <img src="https://forum.endeavouros.com/images/emoji/twitter/slight_smile.png?v=12" title=":slight_smile:" class="emoji" alt=":slight_smile:" loading="lazy" width="20" height="20"></p>
<p>Cheers !!</p>
            </div>

**Comment by manorit2001** [comment]
Hi @LaMarMot , I think it's working as expected, you can actually see that the directory that you have would be a mountpoint actually and not a symlink, this change was made in asd with 6.0 version to make the user experience more transparent and help with the scenarios where asd crashes and restoration introduces a time gap that leads to the applications crashing.

**Comment by LaMarMot** [comment]
Hey @manorit2001, thanks a lot for your answer (and for your work by the way !).

I kind of remembered `asd` was using simlinks before (and is still referencing it in its description here by the way), and, as I'm not familiar with "bind mounting" (which I only discovered thanks to you :sweat_smile:) I then started to wonder what was wrong when I realize the directories weren't simlinks.

Thanks a lot for your reply and insights, much appreciated!
